src/Microsoft.Data.Sqlite.Core/SqliteDataReader.cs

```diff
@@ -373,7 +373,23 @@ public override byte GetByte(int ordinal)
         /// <param name="ordinal">The zero-based column ordinal.</param>
         /// <returns>The value of the column.</returns>
         public override char GetChar(int ordinal)
-            => (char)GetInt64(ordinal);
+        {
+            var sqliteType = GetSqliteType(ordinal);
+            switch (sqliteType)
+            {
+                case raw.SQLITE_FLOAT:
+                case raw.SQLITE_INTEGER:
+                    return (char)GetInt64(ordinal);
+                default:
+                    var val = GetString(ordinal);
```

Comment:
I think we should flip this. Only call `GetString()` if it's `SQLITE_TEXT` otherwise default to the current behavior.